src/flows/components/header/Submit.tsx

```diff
@@ -16,130 +9,64 @@ import {
 import {QueryContext} from 'src/flows/context/query'
-import {FlowContext} from 'src/flows/context/flow.current'
 import {RunModeContext, RunMode} from 'src/flows/context/runMode'
-import {ResultsContext} from 'src/flows/context/results'
 import {notify} from 'src/shared/actions/notifications'
-import {PIPE_DEFINITIONS} from 'src/flows'
+import {FlowContext} from 'src/flows/context/flow.current'
 
 // Utils
-import {event} from 'src/cloud/utils/reporting'
 import {cancelAllRunningQueries} from 'src/timeMachine/actions/queries'
+import {event} from 'src/cloud/utils/reporting'
 
-// Constants
-import {
-  notebookRunSuccess,
-  notebookRunFail,
-} from 'src/shared/copy/notifications'
-import {PROJECT_NAME} from 'src/flows'
+// Styles
+import 'src/flows/components/header/Submit.scss'
 
 // Types
 import {RemoteDataState} from 'src/types'
 
-// Styles
-import 'src/flows/components/header/Submit.scss'
-
 const fakeNotify = notify
 
 export const Submit: FC = () => {
-  const dispatch = useDispatch()
-  const {query, generateMap} = useContext(QueryContext)
-  const {runMode, setRunMode} = useContext(RunModeContext)
   const {flow} = useContext(FlowContext)
-  const {add, update} = useContext(ResultsContext)
-  const [isLoading, setLoading] = useState(RemoteDataState.NotStarted)
-
-  const hasQueries = useMemo(() => {
-    return !!flow.data.all
-      .map(p => PIPE_DEFINITIONS[p.type])
-      .filter(p => p && (p.family === 'transform' || p.family === 'inputs'))
-      .length
-  }, [flow.data])
-
-  // Returns a string to avoid array deep comparison woes
-  // Triggers a re-run any time any aggregate function is changed
-  const aggregateOfAggregates = useMemo(() => {
-    return flow.data.all
-      .map(p => p.aggregateFunction?.name)
-      .filter(p => !!p)
-      .join('')
-  }, [flow.data])
+  const {runMode, setRunMode} = useContext(RunModeContext)
+  const {generateMap, queryAll} = useContext(QueryContext)
 
   const _range = useMemo(
     () => `${flow.range.lower} to ${flow.range.upper || 'now'}`,
     [flow.range]
   )
 
+  const hasQueries = useMemo(() => {
+    return generateMap().length > 0
+  }, [generateMap])
+
   useEffect(() => {
     if (hasQueries) {
-      submit()
+      queryAll()
     }
-  }, [_range, hasQueries, aggregateOfAggregates]) // eslint-disable-line react-hooks/exhaustive-deps
+  }, [_range, hasQueries]) // eslint-disable-line react-hooks/exhaustive-deps
 
-  const forceUpdate = (id, data) => {
-    try {
-      update(id, data)
-    } catch (_e) {
-      add(id, data)
-    }
+  const handleSubmit = () => {
+    event('Notebook Submit Button Clicked')
+    queryAll()
   }
 
-  const submit = () => {
-    if (isLoading === RemoteDataState.Loading) {
-      return
-    }
-
-    const map = generateMap(runMode === RunMode.Run)
-
-    if (!map.length) {
-      return
+  const status = flow.meta.all.reduce((_, curr) => {
+    switch (curr?.loading) {
```

Comment:
we gotta get a little more complicated with this one. as is, this'll only return the loading state of the last pipe. instead, lets change it back to an `isLoading` boolean, but write the reduce like this:
```
const isLoading = flow.meta.all.reduce((acc, curr) => {
  return acc || curr.loading === RemoteDataState.Loading
}, false)
```
with this, as long as any of them are still loading, no mater how they were triggered or how long they take, the submit button stays loading